core/src/main/scala/kafka/admin/PreferredReplicaLeaderElectionCommand.scala

```diff
@@ -119,32 +118,21 @@ object PreferredReplicaLeaderElectionCommand extends Logging {
   }
 }
 
-class PreferredReplicaLeaderElectionCommand(zkUtils: ZkUtils, partitions: scala.collection.Set[TopicAndPartition])
+class PreferredReplicaLeaderElectionCommand(zkUtils: ZkUtils, partitionsFromUser: scala.collection.Set[TopicAndPartition])
   extends Logging {
   def moveLeaderToPreferredReplica() = {
     try {
-      val validPartitions = partitions.filter(p => validatePartition(zkUtils, p.topic, p.partition))
+      val topics = partitionsFromUser.map(_.topic).toSet
+      val partitionsFromZk = zkUtils.getPartitionsForTopics(topics.toSeq).flatMap{case (topic, partitions) =>
+        partitions.map(TopicAndPartition(topic, _))}.toSet
+
+      val (validPartitions, invalidPartitions) = partitionsFromUser.partition(p => partitionsFromZk.contains(p))
```

Comment:
Nit: you can simply write: `partitionsFromUser.partition(partitionsFromZk.contains)`.